derive/src/traits.rs

```diff
@@ -1215,7 +1248,7 @@ impl<'a, I: Iterator<Item = &'a Variant> + 'a> Iterator
       self.last_value += 1;
```

Comment:
⚠️ Unrelated to this PR, but it seems rather important (somebody compiling proc-macros with a different choice of the `overflow-checks` knob may run into a panic with this logic):

```suggestion
      self.last_value = self.last_value.wrapping_add(1);
```

Reply:
Hmm, in the case of the (unstable) `repr128` feature, this could cause unsoundness with `derive(CheckedBitPattern)` on an `enum X { A = i64::MAX as i128, B }`, so to be safe, I'll make `last_value` an `i128` and then do the `wrapping_add` (since the compiler will catch any overflows at the actual discriminant type).